client/post-editor/controller.js

```diff
@@ -91,13 +92,21 @@ function maybeRedirect( context, postType, site ) {
 	return false;
 }
 
-function getPressThisContent( text, url, title ) {
-	return ReactDomServer.renderToStaticMarkup(
-		<p>
-			{ text ? <blockquote>{ text }</blockquote> : null }
-			via <a href={ url }>{ title }</a>.
-		</p>
-	);
+function getPressThisContent( text, url, title, image, embed ) {
+	let pieces = [];
+	if ( image ) {
+		pieces.push( ReactDomServer.renderToStaticMarkup( <p><a href={ url }><img src={ image } /></a></p> ) );
+	}
+	if ( isValidUrl( embed ) ) {
+		pieces.push( embed );
+	}
+	if ( text ) {
+		pieces.push( ReactDomServer.renderToStaticMarkup( <blockquote>{ text }</blockquote> ) );
+	}
+
+	pieces.push( ReactDomServer.renderToStaticMarkup( <p>via <a href={ url }>{ title }</a>.</p> ) );
```

Comment:
I don't think so. the post title that we're using passing along has already been de-entified and stripped of tags. I think.

But if someone else tries to pass random stuff, I really don't want it slipping through.